Account/FirefoxAccount.swift

```diff
@@ -181,6 +182,19 @@ open class FirefoxAccount {
         }
     }
 
+    open func updateFxADevices(withDb db: BrowserDB) -> Success {
+        let cachedState = stateCache.value!
+        guard let session = cachedState as? TokenState else {
+            return deferMaybe(FxAClientError.local(NSError()))
+        }
+        let client = FxAClient10(endpoint: self.configuration.authEndpointURL)
+        return client.devices(withSessionToken: session.sessionToken as NSData) >>== { resp in
+            let devices = resp.devices.map { RemoteDevice(id: $0.id!, name: $0.name, type: $0.type!, isCurrentDevice: $0.isCurrentDevice, lastAccessTime: $0.lastAccessTime) }
```

Comment:
Consider implementing a static method for `RemoteDevice` to allow it to be built from the JSON response, rather than inlining that logic here; you should be able to write a test for it.